ts/utils/login.ts

```diff
@@ -84,21 +94,22 @@ export const getIdpLoginUri = (idpId: string, level: number) =>
  */
 export const onLoginUriChanged =
   (
-    onFailure: (errorCode: string | undefined) => void,
-    onSuccess: (_: SessionToken) => void
+    onFailure: (errorCode?: string, errorMessage?: string) => void,
+    onSuccess: (_: SessionToken) => void,
+    idp: "SPID" | "CIEID" | "CIE" = "SPID"
```

Comment:
What about replacing `CIEID` with `CIE_ID`? I think the second one is easier to read. What do you think?